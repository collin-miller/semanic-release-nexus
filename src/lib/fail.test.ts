import { describe, expect, it, jest } from '@jest/globals';
import { fail } from './fail';

describe('fail', () => {
    it('should log the fail message with single artifact', () => {
        const logMock = jest.fn();
        fail({ assets: [{ path: './some-path' }] }, { logger: { log: logMock, error: jest.fn() }, env: {} });
        expect(logMock).toBeCalledTimes(1);
        expect(logMock.mock.calls).toMatchSnapshot();
    });
    it('should log the fail message with multiple artifacts', () => {
        const logMock = jest.fn();
        fail(
            { assets: [{ path: './some-path-again' }, { path: 'some-artifact' }] },
            { logger: { log: logMock, error: jest.fn() }, env: {} },
        );
        expect(logMock).toBeCalledTimes(1);
        expect(logMock.mock.calls).toMatchSnapshot();
    });
});
