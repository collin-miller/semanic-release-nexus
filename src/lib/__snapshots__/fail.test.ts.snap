// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fail should log the fail message with multiple artifacts 1`] = `
Array [
  Array [
    "Failed to upload artifact(s): ./some-path-again and some-artifact.",
  ],
]
`;

exports[`fail should log the fail message with single artifact 1`] = `
Array [
  Array [
    "Failed to upload artifact(s): ./some-path.",
  ],
]
`;
